src/collections/componentSet.ts

```diff
@@ -128,15 +131,15 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
     const resolver = new MetadataResolver(registry, tree);
     const set = new ComponentSet([], registry);
-    const buildComponents = (paths: string[], asDeletes: boolean): void => {
+    const buildComponents = (paths: string[], destructiveType?: DestructiveChangesType): void => {
       for (const path of paths) {
         for (const component of resolver.getComponentsFromPath(path, inclusiveFilter)) {
-          set.add(component, asDeletes);
+          set.add(component, destructiveType);
```

Comment:
Just dbl-checking here, as I'm not clear on the code paths - since `destructiveType` is nullable, what happens when `set.add(component, null)` is called and then later used?

Reply:
it should default to a `POST` i.e. generating a `destructiveChangesPost.xml`